api/src/main/java/openconsensus/trace/SpanId.java

```diff
@@ -53,36 +54,31 @@ private SpanId(long id) {
   }
 
   /**
-   * Returns a {@code SpanId} whose representation is copied from the {@code src} beginning at the
-   * {@code srcOffset} offset.
+   * Returns a {@code SpanId} whose representation is copied from the {@code src}.
    *
    * @param src the buffer where the representation of the {@code SpanId} is copied.
-   * @param srcOffset the offset in the buffer where the representation of the {@code SpanId}
-   *     begins.
    * @return a {@code SpanId} whose representation is copied from the buffer.
    * @throws NullPointerException if {@code src} is null.
-   * @throws IndexOutOfBoundsException if {@code srcOffset+SpanId.SIZE} is greater than {@code
-   *     src.length}.
+   * @throws IndexOutOfBoundsException if {@code SpanId.SIZE} is greater than {@code
+   *     src.remaining()}.
    * @since 0.1.0
    */
-  public static SpanId fromBytes(byte[] src, int srcOffset) {
+  public static SpanId fromBytes(ByteBuffer src) {
     Utils.checkNotNull(src, "src");
-    return new SpanId(BigendianEncoding.longFromByteArray(src, srcOffset));
+    return new SpanId(BigendianEncoding.longFromByteBuffer(src));
   }
 
   /**
-   * Copies the byte array representations of the {@code SpanId} into the {@code dest} beginning at
-   * the {@code destOffset} offset.
+   * Copies the byte array representations of the {@code SpanId} into the {@code dest}.
    *
    * @param dest the destination buffer.
-   * @param destOffset the starting offset in the destination buffer.
    * @throws NullPointerException if {@code dest} is null.
-   * @throws IndexOutOfBoundsException if {@code destOffset+SpanId.SIZE} is greater than {@code
-   *     dest.length}.
+   * @throws IndexOutOfBoundsException if {@code SpanId.SIZE} is greater than {@code
+   *     dest.remaining()}.
    * @since 0.1.0
    */
-  public void copyBytesTo(byte[] dest, int destOffset) {
-    BigendianEncoding.longToByteArray(id, dest, destOffset);
+  public void copyBytesTo(ByteBuffer dest) {
+    BigendianEncoding.longToByteBuffer(id, dest);
```

Comment:
You cannot make the assumption that this ByteBuffer instance has the big endian order set. So probably you should make sure is big-endian:
```java
oldOrder = dest.order();
dest.setOrder(ByteOrder.BIG_ENDIAN);
BigendianEncoding.longToByteBuffer(id, dest);
dest.setOrder(oldOrder);
```

Same for fromBytes and TraceId.

Reply:
Was thinking about it a few days ago - in a perfect world we would expect a big-endian `ByteBuffer`, but I guess it's not so good to take that for granted.

So I will indeed read the long as big-endian and will add a note on it in the inline docs.